core/utils/queue.h

```diff
@@ -0,0 +1,24 @@
+#ifndef BESS_UTILS_QUEUE_H
+#define BESS_UTILS_QUEUE_H
+
+namespace bess {
+namespace utils {
+
+// Takes Template argument T that is the type to be enqueued and dequeued.
+template <typename T>
+class Queue {
+ public:
+  virtual ~Queue(){};
+  virtual int Push(T*) = 0;
```

Comment:
1. Seems infeasible, since you cannot create a dummy object to return if T is not [default constructible](http://en.cppreference.com/w/cpp/concept/DefaultConstructible).
2. sounds good. The return type could be either bool or int (0 or -errno).
3. Like it or not, the Google C++ Style Guide does not allow using exceptions.